### PR TITLE
Support for json file generation

### PR DESCRIPTION
**Pull request by egbaquela**
I added support for output in json format. For that, I added json.hpp library. Now, there are a new command line option, -j or --jsonFile, which indicates the filename of the file in which the json output will be wrote.
Json output is not rendered on screen, it can only be wrote to files.

### CODE REVIEW

File: main.cpp [hunk 1]
```diff
@@ -285,9 +320,13 @@ int main(int argc, char** argv)
     /// Assign Pareto optimal values
     if(prob == 5) {
         cout << "\n% Optimal set cannot be analytically derived for CODeM5" << endl;
+        if (writeJsonFile){
+            jsonOutput["status"] = "Optimal set cannot be analytically derived for CODeM5";
+        }
     }
     else
     {
+        jsonOutput["status"] = "Optimal set can be analytically derived for CODeM5";
```
Comment (ShaulSalomon) [inline main.cpp]: 'else' means it is not CODeM5, and therefore the optimal set can be analytically derived.
The 'status' is not correct here

Comment (ShaulSalomon) [inline main.cpp:329]: 'else' means it is not CODeM5, and therefore the optimal set can be analytically derived.
The 'status' is not correct here

File: main.cpp [hunk 2]
```diff
@@ -285,9 +320,13 @@ int main(int argc, char** argv)
     /// Assign Pareto optimal values
     if(prob == 5) {
         cout << "\n% Optimal set cannot be analytically derived for CODeM5" << endl;
+        if (writeJsonFile){
+            jsonOutput["status"] = "Optimal set cannot be analytically derived for CODeM5";
```
Comment (ShaulSalomon) [inline main.cpp:324]: Should this status be used by the program that reads the json file? Maybe it should be a Boolean that states whether the optimal set can be derived?

**Review by ShaulSalomon — CHANGES_REQUESTED** [review]
Thanks for your contribution.
This seems ok. I only cannot understand the 'status' field.
If it is required, maybe consider changing it to a flag, otherwise, please consider removing it.

File: main.cpp (hunk 2 above)
Comment (egbaquela) [reply main.cpp:324]: It is really not used. I will delete all references to status.

File: main.cpp (hunk 1 above)
Comment (egbaquela) [reply main.cpp:329]: I will delete it

**Pull request closed without merging**